### PR TITLE
Updating Functions V4 item templates to use newest extension packages

### PR DESCRIPTION
**Pull request by surgupta-msft**
Resolves https://github.com/Azure/azure-functions-templates/issues/1097 
Spec - https://github.com/Azure/Azure-Functions/issues/1987

Details of the versions upgraded -
Microsoft.Azure.WebJobs.Extensions.Storage | 4.0.5
Microsoft.Azure.WebJobs.Extensions.ServiceBus | 4.3.0
Microsoft.Azure.WebJobs.Extensions.EventHubs | 4.2.0
Microsoft.Azure.WebJobs.Extensions.DurableTask | 2.5.1
Microsoft.Azure.WebJobs.Extensions.CosmosDB | 3.0.10
Microsoft.Azure.WebJobs.Extensions.EventGrid | 2.1.0
Microsoft.Azure.WebJobs.Extensions.RabbitMQ | 1.1.0
Microsoft.Azure.WebJobs.Extensions.Kafka | 3.3.2

### CODE REVIEW

**Review by fabiocav — APPROVED** [review]
Change looks good to me, but it would be good to get an ack from @soninaren, particularly with the major version bumps

File: Functions.Templates/Templates/DurableFunctionsActivity-CSharp-1.x/metadata.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
   "userPrompt": [],
   "extensions": [
     {
-      "id": "Microsoft.Azure.WebJobs.Extensions.DurableTask", "version": "1.8.7"
+      "id": "Microsoft.Azure.WebJobs.Extensions.DurableTask", "version": "2.5.1"
```
Comment (soninaren) [inline Functions.Templates/Templates/DurableFunctionsActivity-CSharp-1.x/metadata.json:15]: This is incorrect. This template specifically target v1 of durable. Updating the extension version will break the template. Also i don't think this template is currently being used.

File: Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-1.x/.template.config/template.json [hunk 2]
```diff
@@ -34,7 +34,7 @@
       "ManualInstructions": [],
       "args": {
         "referenceType": "package",
-        "reference": "Microsoft.Azure.WebJobs.Extensions.DurableTask", "version": "1.8.7",
+        "reference": "Microsoft.Azure.WebJobs.Extensions.DurableTask", "version": "2.5.1",
```
Comment (soninaren) [inline Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-1.x/.template.config/template.json:37]: This should reverted to `1.8.7`

**Review by soninaren — CHANGES_REQUESTED** [review]
cc: @ConnorMcMahon for durable changes

File: Functions.Templates/Templates/DurableFunctionsActivity-CSharp-1.x/metadata.json (hunk 1 above)
Comment (surgupta-msft) [reply Functions.Templates/Templates/DurableFunctionsActivity-CSharp-1.x/metadata.json:15]: @soninaren  Sure. Do you suggest to not update the version for Durable and keep it 1.8.7?

Comment (surgupta-msft) [reply Functions.Templates/Templates/DurableFunctionsActivity-CSharp-1.x/metadata.json:15]: Sorry, got my answer from your another comment.

File: Functions.Templates/Templates/DurableFunctionsHttpStart-CSharp-1.x/metadata.json [hunk 3]
```diff
@@ -14,7 +14,7 @@
   ],
   "extensions": [
     {
-      "id": "Microsoft.Azure.WebJobs.Extensions.DurableTask", "version": "1.8.7"
+      "id": "Microsoft.Azure.WebJobs.Extensions.DurableTask", "version": "2.5.1"
```
Comment (soninaren) [inline Functions.Templates/Templates/DurableFunctionsHttpStart-CSharp-1.x/metadata.json:17]: Need to keep this as is

File: Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-1.x/metadata.json [hunk 4]
```diff
@@ -12,7 +12,7 @@
   "userPrompt": [],
   "extensions": [
     {
-      "id": "Microsoft.Azure.WebJobs.Extensions.DurableTask", "version": "1.8.7"
+      "id": "Microsoft.Azure.WebJobs.Extensions.DurableTask", "version": "2.5.1"
```
Comment (soninaren) [inline Functions.Templates/Templates/DurableFunctionsOrchestration-CSharp-1.x/metadata.json:15]: Need to keep this as is

File: Functions.Templates/Templates/DurableFunctionsOrchestrator-CSharp-1.x/metadata.json [hunk 5]
```diff
@@ -12,7 +12,7 @@
   "userPrompt": [],
   "extensions": [
     {
-      "id": "Microsoft.Azure.WebJobs.Extensions.DurableTask", "version": "1.8.7"
```
Comment (soninaren) [inline Functions.Templates/Templates/DurableFunctionsOrchestrator-CSharp-1.x/metadata.json:15]: Need to keep this as is

**Review by soninaren — CHANGES_REQUESTED** [review]

File: Functions.Templates/Templates/DurableFunctionsHttpStart-CSharp-1.x/metadata.json (hunk 3 above)
Comment (surgupta-msft) [reply Functions.Templates/Templates/DurableFunctionsHttpStart-CSharp-1.x/metadata.json]: updated the code. Thanks for catching it.

**Review by soninaren — APPROVED** [review]

**Pull request merged**